app/controllers/comments_controller.rb

```diff
@@ -28,6 +28,7 @@ class CommentsController < ApplicationController
   before_action :check_permission_to_moderate, only: [:approve, :reject]
   before_action :check_permission_to_modify_frozen_status, only: [:freeze, :unfreeze]
   before_action :check_permission_to_modify_hidden_status, only: [:hide, :unhide]
+  before_action :admin_logout_required, only: [:create]
```

Comment:
I really do think you should add `:new` and `:add_comment_reply` here. Otherwise, an admin who opens one of those pages is going to get this behavior:
1) The new comments page still has "New comment on TITLE", but there's nothing at all on the page.
2) Opening the comments section logged-out in one tab, logging in as admin in another tab, and then clicking "Reply" in the first tab makes the navigation for that comment disappear. (Normally it would make the navigation disappear and replace it with a comment box. But with the comment box hidden, it just inexplicably takes away the Thread/Parent Thread links.)

(In general, any filters that apply to the comment `create` action should probably also be applied to `new` and `add_comment_reply`, unless there's a specific reason that it should apply only when submitting the new comment.)